jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh

```diff
@@ -97,15 +92,53 @@ function configure_router_location {
     fi
     JBOSS_KIE_ARGS="${JBOSS_KIE_ARGS} -Dorg.kie.server.router.port=${kieServerRouterPort}"
 
+    local kieServerRouterProtocol="${KIE_SERVER_ROUTER_PROTOCOL}"
+    if [ "${kieServerRouterProtocol}" = "" ]; then
+        kieServerRouterProtocol="http"
+    fi
+
     local kieServerRouterUrlExternal="${KIE_SERVER_ROUTER_URL_EXTERNAL}"
+    local routeName="${KIE_SERVER_ROUTER_NAME// /_}"
+    local kieServerRouterAppName="${KIE_SERVER_ROUTER_APPLICATION_NAME}"
+    
+    # value from  KIE_SERVER_ROUTER_NAME (org.kie.server.router.name) if no route name is defined
+    if [ "${routeName}" = "KIE_Server_Router" ]; then
+        if [ "${kieServerRouterProtocol}" = "https" ]; then
+            routeName=secure-"${kieServerRouterAppName}"-smartrouter
+        else
+            routeName="${kieServerRouterAppName}"-smartrouter
+        fi
+    fi
+    
     if [ "${kieServerRouterUrlExternal}" = "" ]; then
-        local kieServerRouterProtocol="${KIE_SERVER_ROUTER_PROTOCOL}"
-        if [ "${kieServerRouterProtocol}" = "" ]; then
-            kieServerRouterProtocol="http"
+        # only execute the following lines if this container is running on OpenShift
```

Comment:
don't replicate code, this specific function is under [jboss-kie-wildfly-common](https://github.com/jboss-container-images/jboss-kie-modules/blob/master/jboss-kie-wildfly-common/added/launch/jboss-kie-wildfly-common.sh#L47) module, you can import it with
`source "${JBOSS_HOME}/bin/launch/jboss-kie-wildfly-common.sh"`

Reply:
@spolti , the problem with re-using that script is that smartrouter is not a wildfly (eap) image. It is just an executable jar. So perhaps if we don't want to replicate code, we should create a new jboss-kie-common module, which can be used both by wildfly and non-wildfly images. Thoughts?